pkg/agent/cniserver/sriov.go

```diff
@@ -0,0 +1,153 @@
+package cniserver
+
+import (
+	"fmt"
+	"context"
+	"net"
+	"os"
+	"path"
+	"time"
+	"google.golang.org/grpc"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"antrea.io/antrea/pkg/agent/util"
+	podresourcesv1alpha1 "k8s.io/kubelet/pkg/apis/podresources/v1alpha1"
+	sriovcniutils "github.com/k8snetworkplumbingwg/sriov-cni/pkg/utils"
+	"k8s.io/klog/v2"
+)
+
+const (
+	kubeletPodResourcesPath = "/var/lib/kubelet/pod-resources"
+	kubeletSocket           = "kubelet.sock"
+	connectionTimeout       = 10 * time.Second
+	maxMsgSize              = 1024 * 1024 * 16 // 16 Mb
+)
+
+type KubeletPodResources struct {
+	resources []*podresourcesv1alpha1.PodResources
+}
+
+// K8sArgs is the valid CNI_ARGS used for Kubernetes.
+type K8sArgs struct {
+	types.CommonArgs
+	IP                         net.IP
+	K8S_POD_NAME               types.UnmarshallableString
+	K8S_POD_NAMESPACE          types.UnmarshallableString
+	K8S_POD_INFRA_CONTAINER_ID types.UnmarshallableString
+}
+
+func GetPodResourceRequests(podName string, podNamespace string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), connectionTimeout)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, path.Join(kubeletPodResourcesPath, kubeletSocket),
+		grpc.WithInsecure(), grpc.WithContextDialer(func(ctx context.Context, addr string) (conn net.Conn, e error) {
+			return util.DialLocalSocket(addr)
+		}))
+	if err != nil {
+		return "", fmt.Errorf("Error getting the gRPC client for pod resources: %v", err)
```

Comment:
Error -> error - we do not capitalize first letter in an error message.

Reply:
Done